cmd/pull.go

```diff
@@ -0,0 +1,52 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package cmd
+
+import (
+	"fmt"
+
+	"github.com/xataio/pgroll/cmd/flags"
+	"github.com/xataio/pgroll/pkg/state"
+
+	"github.com/spf13/cobra"
+)
+
+func pullCmd() *cobra.Command {
+	opts := map[string]string{
+		"t": "prepend each migration filename with its timestamp",
+	}
+	var withTimestamps bool
+
+	pullCmd := &cobra.Command{
+		Use:   "pull <target directory>",
+		Short: "pull migration history from the target database and write it to disk",
+		Args:  cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ctx := cmd.Context()
+			targetDir := args[0]
+
+			state, err := state.New(ctx, flags.PostgresURL(), flags.StateSchema())
+			if err != nil {
+				return err
+			}
+			defer state.Close()
+
+			migs, err := state.SchemaHistory(ctx, flags.Schema())
+			if err != nil {
+				return fmt.Errorf("failed to read schema history: %w", err)
+			}
+
+			for _, mig := range migs {
+				err := mig.WriteToFile(targetDir, withTimestamps)
+				if err != nil {
+					return fmt.Errorf("failed to write migration %q: %w", mig.Migration.Name, err)
+				}
+			}
+			return nil
+		},
+	}
+
+	pullCmd.Flags().BoolVarP(&withTimestamps, "with-timestamps", "t", false, opts["t"])
```

Comment:
I've been thinking about this one, what do you think about relying on numbers here? we could do `0001_` etc, looking at the example output, timestamps are not the most readable thing

Reply:
I think you're right.

The flag is now `--with-prefixes` and prefixes each filename with `0001_` etc as you suggest.

PR description and docs are updated to reflect the change.